File: src/main/java/com/example/airmed/Entity/Psychiatrist.java
package com.example.airmed.Entity;

import com.example.airmed.Hashed;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashMap;
import java.util.Map;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Psychiatrist {
    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false, unique = true)
    private String medicalNumber;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String mail;

    @Column(nullable = false, unique = true)
    private String phone;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String country;

    @Column(nullable = false)
    private String locality;

    @Column
    private String cabinetLocation;

    @Column
    private String linkLocation;

    @Column(nullable = false)
    private int priceConsult;

    @Column(nullable = false)
    private int priceConsultation;

    @Column(columnDefinition = "boolean default false")
    @JsonProperty
    private boolean online;

    @Column(columnDefinition = "boolean default false")
    @JsonProperty
    private boolean CNAS;

//    @ElementCollection
//    private Map<String,String> salts = new HashMap<>();
//
//    public void setMedicalNumber(String medicalNumber) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("medicalNumber", salt);
//        this.medicalNumber = Hashed.createHashData(medicalNumber,salt);
//    }
//
//    public void setFirstName(String firstName) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("firstName",salt);
//        this.firstName = Hashed.createHashData(firstName,salt);
//    }
//
//    public void setLastName(String lastName) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("lastName",salt);
//        this.lastName = Hashed.createHashData(lastName,salt);
//    }
//
//    public void setMail(String mail) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("mail",salt);
//        this.mail = Hashed.createHashData(mail,salt);
//    }
//
//    public void setPhone(String phone) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("phone",salt);
//        this.phone = Hashed.createHashData(phone,salt);
//    }
//
//    public void setPassword(String password) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("password",salt);
//        this.password = Hashed.createHashData(password,salt);
//    }
//
//    public void setBio(String bio) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("bio",salt);
//        this.bio = Hashed.createHashData(bio,salt);
//    }
//
//    public void setCountry(String country) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("country",salt);
//        this.country = Hashed.createHashData(country,salt);
//    }
//
//    public void setLocality(String locality) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("locality",salt);
//        this.locality = Hashed.createHashData(locality,salt);
//    }
//
//    public void setCabinetLocation(String cabinetLocation) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("cabinetLocation",salt);
//        this.cabinetLocation = Hashed.createHashData(cabinetLocation,salt);
//    }
//
//    public void setLinkLocation(String linkLocation) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("linkLocation",salt);
//        this.linkLocation = Hashed.createHashData(linkLocation,salt);
//    }
}

File: frontend/src/classes/Psychiatrist.ts
export class Psychiatrist{

    id: number;
    medicalNumber: string;
    firstName: string;
    lastName: string;
    mail: string;
    phone: string;
    country: string;
    locality: string;
    cabinetLocation: string;
    linkLocation: string;
    priceConsult: number;
    priceConsultation: number;
    online: boolean;
    CNAS: boolean;

    constructor(id: number,
                medicalNumber: string,
                firstName: string,
                lastName: string,
                mail: string,
                phone: string,
                country: string,
                locality: string,
                cabinetLocation: string,
                linkLocation: string,
                priceConsult: number,
                priceConsultation: number,
                online: boolean,
                CNAS: boolean) {
        this.id = id;
        this.priceConsultation = priceConsultation;
        this.priceConsult = priceConsult;
        this.CNAS = CNAS;
        this.online = online;
        this.medicalNumber = medicalNumber;
        this.firstName = firstName;
        this.lastName = lastName;
        this.mail = mail;
        this.phone = phone;
        this.country = country;
        this.locality = locality;
        this.cabinetLocation = cabinetLocation;
        this.linkLocation = linkLocation;
    }

    static jsonToPsychiatrist(json: string) {
        const data = JSON.parse(json);
        // console.log("Data ", data);
        return new Psychiatrist(
            data.id,
            data.medicalNumber,
            data.firstName,
            data.lastName,
            data.mail,
            data.phone,
            data.country,
            data.locality,
            data.cabinetLocation,
            data.linkLocation,
            data.priceConsult,
            data.priceConsultation,
            data.online,
            data.CNAS
        );
    }
}

export default function psychiatristToJson(psychiatrist: Psychiatrist): string {
    const json = JSON.stringify({
        id: psychiatrist.id,
        medicalNumber: psychiatrist.medicalNumber,
        firstName: psychiatrist.firstName,
        lastName: psychiatrist.lastName,
        mail: psychiatrist.mail,
        phone: psychiatrist.phone,
        country: psychiatrist.country,
        locality: psychiatrist.locality,
        cabinetLocation: psychiatrist.cabinetLocation,
        linkLocation: psychiatrist.linkLocation,
        priceConsult: psychiatrist.priceConsult,
        priceConsultation: psychiatrist.priceConsultation,
        online: psychiatrist.online,
        CNAS: psychiatrist.CNAS
    });
    return json;
}